include/boost/astronomy/coordinate/base_differential.hpp

```diff
@@ -116,6 +64,19 @@ struct base_differential
         return this->diff;
     }
 
+
+    //! returns the differential of calling object
+    bg::model::point
+    <
+        CoordinateType,
+        DimensionCount,
+        CoordinateSystem
+    >
+    get_point() const
```

Comment:
@RohitRanjangit  you told that `get_point` will help to calculate cross for two differential classes or one representation class and one differential class. But the code segment you're talking about uses `make_cartesian_representation()` so differential can't be an argument for that function: [link to that function](https://github.com/BoostGSoC19/astronomy/blob/175f0b246f37ebb427d6bef99ef8448f8fdd92e2/include/boost/astronomy/coordinate/arithmetic.hpp#L184).
Rest all looks good to me.

Reply:
We can use differential as an argument because of make_cartesian_differential works exactly the same as make_cartesian_representation. The only difference is the former creates a representation class and the latter creates a differential class. If we see the implementations, both classes of cartesian_representation and cartesian_differential have been implemented in the same way (That's why I guess we're able to get correct results if we try cross_product of two differentials).
Anyway, we are creating two temporary cartesian points for calculating the cross_product. So we should not worry about this.

Reply:
I'll update this PR. thanks for pointing out though.